Document `core.EMERGE_*` symbols (#16848)
minetest/minetest: doc/lua_api.md

        * `blockpos` is the *block* coordinates of the block that had been
          emerged.
        * `action` could be one of the following constant values:
            * `core.EMERGE_GENERATED`: The block has been freshly generated
            * `core.EMERGE_FROM_MEMORY`: The block was generated before and has been

>               loaded from transient memory

Reviewer: ```suggestion
            * `core.EMERGE_FROM_MEMORY`: The block did not not need to be loaded,
              because it was already in memory.
```

            * `core.EMERGE_FROM_DISK`: The block was generated before and has been

>               loaded from disk / storage device

Reviewer: ```suggestion
            * `core.EMERGE_FROM_DISK`: The block was generated before and has been
              loaded from the map database.
```

            * `core.EMERGE_CANCELLED`: The block was not loaded for some reason.
              Possible reasons include: Server shutdown, it was already queued

>               to be emerged. This event is not an error

Reviewer: ```suggestion
              Possible reasons include: server shutdown, it was already being emerged,
              outside of mapgen limits
              This event is not an error, but does *not* mean that the block is now present.
```
